### PR TITLE
Upgrade k8s client to v0.16.8

### PR DESCRIPTION
**Pull request by mkmik**
And regenerate client.

The new client has the UpdateStatus method which will be used to address #346

### CODE REVIEW

File: pkg/apis/sealed-secrets/v1alpha1/register.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ var (
 
 	// SchemeBuilder adds this group to scheme
 	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
+	// AddToScheme is a global function that registers this API group & version to a scheme
+	AddToScheme = SchemeBuilder.AddToScheme
```
Comment (atomatt) [inline pkg/apis/sealed-secrets/v1alpha1/register.go]: Is this used? Would it make sense to split it out to a different PR?

File: pkg/apis/sealed-secrets/v1alpha1/doc.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // go mod vendor doesn't preserve executable perm bits
-//go:generate bash ../../../../vendor/k8s.io/code-generator/generate-groups.sh all github.com/bitnami-labs/sealed-secrets/pkg/client github.com/bitnami-labs/sealed-secrets/pkg/apis sealed-secrets:v1alpha1 --go-header-file boilerplate.go.txt
+//go:generate bash ../../../../vendor/k8s.io/code-generator/generate-groups.sh deepcopy,defaulter,client,lister github.com/bitnami-labs/sealed-secrets/pkg/client github.com/bitnami-labs/sealed-secrets/pkg/apis sealed-secrets:v1alpha1 --go-header-file boilerplate.go.txt
```
Comment (atomatt) [inline pkg/apis/sealed-secrets/v1alpha1/doc.go]: I assume this was necessary due to an updated `generate-groups.sh` script? It looks like not using `all` means no `informer` generator, is that correct?

Comment (mkmik) [reply pkg/apis/sealed-secrets/v1alpha1/doc.go]: yeah, the generated informer is buggy: it generates the file as `package sealed-secret` which then fails to gofmt and aborts.

I created this PR mostly to make use of the CI integration test to check I'm on the right track.

I'll find a way to correctly generate the informer with the new version of the generator before merging this

File: pkg/apis/sealed-secrets/v1alpha1/register.go (hunk 1 above)
Comment (mkmik) [reply pkg/apis/sealed-secrets/v1alpha1/register.go]: it seems that github swallowed my answers.

k8s code-generator requires some boilerplate; this is now part of that boilerplate

Comment (atomatt) [reply pkg/apis/sealed-secrets/v1alpha1/register.go]: +1 thanks for checking
